templates/funcs.go

```diff
@@ -264,8 +297,21 @@ func render(f pgs.Field) string {
 				o.AddObject("%s", %s)
 			}`, getter(n, t), name(f), getter(n, t))
 		}
+	} else if t.IsMap() {
+		d := newMapData(f, obsType)
+
+		tpl := template.New("map")
+		template.Must(tpl.Parse(mapTpl))
+		bb := bytes.NewBufferString("")
+		_ = tpl.Execute(bb, d)
+
+		s = bb.String()
+
 	} else {
-		s = fmt.Sprintf(`o.%s("%s", %s)`, simpleAddFunc(n, t), name(f), getter(n, t))
+		if obsType == zap.ObfuscationType_STARS {
+			return fmt.Sprintf("\no.AddString(\"%s\", \"***\")\n", name(f))
```

Comment:
Same here, are you sure you can return? It looks like all other places assign to the `s` variable instead